### PR TITLE
[react-router-dom] Add tests and update lowest supported TypeScript version

### PR DESCRIPTION
**Pull request by galuszkak**
- [X] Use a meaningful title for the pull request. Include the name of the package modified
- [X] Test the change in your own code. (Compile and run.)
- [X] Add or edit tests to reflect the change. (Run with `npm test`.)
- [X] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [X] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Last one is failing and I don't know how to fix that.

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [X] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

File: types/react-router-dom/index.d.ts
Comment (eps1lon) [inline types/react-router-dom/index.d.ts:10]: I'm not so sure about this one. It's technically a breaking change. But I wouldn't know how this applies practially. Since you didn't change any typings I don't see why we need to change it.

**Review by eps1lon — CHANGES_REQUESTED** [review]

File: types/react-router-dom/index.d.ts
Comment (galuszkak) [reply types/react-router-dom/index.d.ts:10]: Hey @eps1lon ,

My thought process was that because to make things up to do date, we should up the version of Typescript to only ones that are supported, so choose min(supportedTypescripts). If that's not the case I'm happy to remove that change. This is to ensure none fills the bug against non supported typescript version.

Comment (eps1lon) [reply types/react-router-dom/index.d.ts:10]: > This is to ensure none fills the bug against non supported typescript version.

But 2.8 is currently supported. I'm not aware of bug reports that only apply to 2.8

Comment (galuszkak) [reply types/react-router-dom/index.d.ts:10]: @eps1lon Can you then explain me what README below means? I thought this is supported versions:
Screenshot 2020-09-12 at 12 27 36 PM

It clearly states that support for 2.8 ended in March 2020. If that's not the case where I can find correct information?

Comment (sandersn) [reply types/react-router-dom/index.d.ts:10]: DT maintainer here: Only 3.1 and above are supported; you don't need this comment anymore if you support 3.1 and above.

(and, uh, looking at that schedule, looks like I should remove 3.1 this week too.)

Comment (eps1lon) [reply types/react-router-dom/index.d.ts:10]: >  2.8 was supported, be aware that DT itself no longer tests that 2.8

@sandersn I wasn't aware of that. Shouldn't the test runner throw an error once the underlying types are changed since the comment is misleading?

Comment (sandersn) [reply types/react-router-dom/index.d.ts:10]: Yeah, that's not a bad idea. I filed https://github.com/microsoft/dtslint/issues/309 to track this.

Comment (eps1lon) [reply types/react-router-dom/index.d.ts:10]: Though it's technically not tested anyway I'd still argue to keep the documented TS version at 2.8. The React core team asked us specifically to not make any type related breaking changes for v17. While that's always possible with types it's less likely if we keep a human check for the kind of TS we allow (using `unknown` for example).

Especially since changing the documented, minimum supported TS version doesn't unblock anything as far as I can tell.

Comment (rbuckton) [reply types/react-router-dom/index.d.ts:10]: @eps1lon Can you clarify this:
> The React core team asked us specifically to not make any type related breaking changes for v17.

Does the change on [Line 62 below](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/47532/files#diff-5e9e7e642af8d89c125bd50d8354b670R62) constitute a type-related breaking change that should be avoided? If so, should this PR not be merged with that change?

Comment (eps1lon) [reply types/react-router-dom/index.d.ts:10]: Oh no, in my mind we were talking about `react` not `react-router-dom`. Discard all that was said. It applies only to `react` not `react-router-dom`

**Review by eps1lon — APPROVED** [review]

**Review by sandersn — APPROVED** [review]

**Comment by galuszkak** [comment]
Ready to merge

**Pull request merged**
